### PR TITLE
add: setup all schemas frontend

### PR DESCRIPTION
**Pull request by thanglp163**

### CODE REVIEW

File: frontend/src/schemas/booking.schema.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { User } from '@/schemas/user.schema';
+import { Payment } from '@/schemas/payment.schema';
+import { Party } from '@/schemas/party.schema';
+
+export interface Booking {
+  _id: string;
+  user: User[];
```
Comment (bakaqc) [inline frontend/src/schemas/booking.schema.ts]: type of user is User not User[ ]

File: frontend/src/schemas/booking.schema.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  party?: Party[];
```
Comment (bakaqc) [inline frontend/src/schemas/booking.schema.ts]: type of party is Party not Party[ ]

**Review by theqi106 — APPROVED** [review]

File: frontend/src/schemas/notify.schema.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { User } from '@/schemas/user.schema';
+
+export interface Notify {
+	_id: string;
+	user: User;
```
Comment (bakaqc) [inline frontend/src/schemas/notify.schema.ts]: type of user is User[ ] not User

File: frontend/src/schemas/party.schema.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { User } from '@/schemas/user.schema';
+
+export interface Party {
+	_id: string;
+	user: User;
+	category: string;
```
Comment (bakaqc) [inline frontend/src/schemas/party.schema.ts]: false
follow: enum: ['Sinh nhật', 'Đám cưới', 'Khai trương', 'Thôi nôi'],

File: frontend/src/schemas/party.schema.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+	title: string;
+	description: string;
+	options: 'tiệc cưới' | 'sinh nhật' | 'thôi nôi' | 'khai trương';
```
Comment (bakaqc) [inline frontend/src/schemas/party.schema.ts]: false
follow: 
type: {
					type: String,
					enum: ['Basic', 'Premium', 'VIP'],
					required: true,
},
price: { type: Number, required: true },

File: frontend/src/schemas/payment.schema.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import { Voucher } from '@/schemas/voucher.schema';
+import { Booking } from '@/schemas/booking.schema';
+
+export interface Payment {
+	_id: string;
+	booking: Booking;
+	voucher?: Voucher[];
```
Comment (bakaqc) [inline frontend/src/schemas/payment.schema.ts]: type of voucher is Voucher not Voucher[ ]

File: frontend/src/schemas/payment.schema.ts [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+	originPrice: number;
+	finalPrice: number;
+	method: 'vn-pay' | 'momo' | 'zalo-pay';
```
Comment (bakaqc) [inline frontend/src/schemas/payment.schema.ts]: method don't have vn-pay

File: frontend/src/schemas/notify.schema.ts (hunk 3 above)
Comment (thanglp163) [reply frontend/src/schemas/notify.schema.ts]: done

File: frontend/src/schemas/booking.schema.ts [hunk 8]
```diff
@@ -0,0 +1,16 @@
+import { User } from '@/schemas/user.schema';
+import { Payment } from '@/schemas/payment.schema';
+import { Party } from '@/schemas/party.schema';
+
+export interface Booking {
+  _id: string;
+  user: User;
+  party?: Party;
```
Comment (bakaqc) [inline frontend/src/schemas/booking.schema.ts]: party is require not optional

File: frontend/src/schemas/booking.schema.ts [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+  guestCount: number;
+  status: 'PENDING' | 'APPROVED' | 'CANCELLED';
+  payment: Payment;
+  organizeDate?: Date;
```
Comment (bakaqc) [inline frontend/src/schemas/booking.schema.ts]: require not optional

File: frontend/src/schemas/booking.schema.ts [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+import { Payment } from '@/schemas/payment.schema';
+import { Party } from '@/schemas/party.schema';
+
+export interface Booking {
+  _id: string;
+  user: User;
+  party?: Party;
+  guestCount: number;
+  status: 'PENDING' | 'APPROVED' | 'CANCELLED';
+  payment: Payment;
```
Comment (bakaqc) [inline frontend/src/schemas/booking.schema.ts]: type of payment is optional not  require

File: frontend/src/schemas/notify.schema.ts (hunk 3 above)
Comment (bakaqc) [reply frontend/src/schemas/notify.schema.ts]: haven't fixed

File: frontend/src/schemas/party.schema.ts [hunk 11]
```diff
@@ -0,0 +1,15 @@
+import { User } from '@/schemas/user.schema';
+
+export interface Party {
+	_id: string;
+	user: User;
+	category: 'Sinh nhật' | 'Đám cưới' | 'Khai trương' | 'Thôi nôi';
+	title: string;
+	description: string;
```
Comment (bakaqc) [inline frontend/src/schemas/party.schema.ts:8]: where is `option` attribute?

File: frontend/src/schemas/party.schema.ts [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+	price: number;
```
Comment (bakaqc) [inline frontend/src/schemas/party.schema.ts]: follow database haven't `price`

File: frontend/src/schemas/user.schema.ts [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+	fullName: string;
+	email: string;
+	hashedPassword: string;
+	phoneNumber: string;
+	address: string;
+	dateOfBirth: Date;
+	gender: 'male' | 'female';
+	avatar: string;
+	role: 'user' | 'admin' | 'super-admin';
+	verificationCode: string;
```
Comment (bakaqc) [inline frontend/src/schemas/user.schema.ts]: optional not require

File: frontend/src/schemas/user.schema.ts [hunk 14]
```diff
@@ -0,0 +1,17 @@
...
+	verificationCodeExpires: Date;
```
Comment (bakaqc) [inline frontend/src/schemas/user.schema.ts]: optional not require

File: frontend/src/schemas/user.schema.ts [hunk 15]
```diff
@@ -0,0 +1,17 @@
...
+	isVerified: boolean;
```
Comment (bakaqc) [inline frontend/src/schemas/user.schema.ts]: optional not require

**Review by bakaqc — CHANGES_REQUESTED** [review]
fix all comment before merge

File: frontend/src/schemas/party.schema.ts [hunk 16]
```diff
@@ -0,0 +1,15 @@
+import { User } from '@/schemas/user.schema';
+
+export interface Party {
+	_id: string;
+	user: User;
+	category: 'Sinh nhật' | 'Đám cưới' | 'Khai trương' | 'Thôi nôi';
+	title: string;
+	description: string;
+	options: string[];
```
Comment (bakaqc) [inline frontend/src/schemas/party.schema.ts]: type of option is false
follow:
options: Option[ ];

export interface Option {
	type: 'Basic' | 'Premium' | 'VIP';
	price: number;
}

**Review by bakaqc — APPROVED** [review]

**Pull request merged**